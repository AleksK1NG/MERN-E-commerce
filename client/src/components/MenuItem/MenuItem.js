import React from 'react'
import { withRouter } from 'react-router-dom'

import {
  MenuItemContainer,
  BackgroundImageContainer,
  ContentContainer,
  ContentSubtitle,
  ContentTitle
} from './MenuItem.styles'

export const MenuItem = ({ title, imageUrl, linkUrl, size, history, match }) => {
  return (
    <MenuItemContainer size={size} onClick={() => history.push(`${match.url}${linkUrl}`)}>
      <BackgroundImageContainer className="background-image" imageUrl={imageUrl} />

      <ContentContainer className="content">
        <ContentTitle>{title.toUpperCase()}</ContentTitle>
        <ContentSubtitle>SHOP NOW</ContentSubtitle>
      </ContentContainer>
    </MenuItemContainer>
  )
}

export default React.memo(withRouter(MenuItem))
