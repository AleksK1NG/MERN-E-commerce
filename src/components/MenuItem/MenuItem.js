import React from 'react'
import './MenuItem.styles.scss'

const MenuItem = ({ title }) => {
  return (
    <div className="menu-item">
      <div className="content">
        <h1 className="title">{title}</h1>
        <span className="subtitle">SHOP NOW</span>
      </div>
    </div>
  )
}

export default React.memo(MenuItem)
