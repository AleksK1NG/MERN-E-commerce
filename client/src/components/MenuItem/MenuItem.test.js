import React from 'react'
import { shallow } from 'enzyme'
import { MenuItem } from './MenuItem'

describe('MenuItem component', () => {
  let wrapper
  let mockMatch
  let mockHistory
  const linkUrl = '/hats'
  const size = 'large'
  const imageUrl = 'testimage'
  const title = 'hats'

  beforeEach(() => {
    mockMatch = {
      url: '/shop'
    }

    mockHistory = {
      push: jest.fn()
    }

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      linkUrl,
      size,
      title,
      imageUrl
    }

    wrapper = shallow(<MenuItem {...mockProps} />)
  })

  it('should render menuItem component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call history.push with the right string when MenuItemContainer clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click')

    expect(mockHistory.push).toHaveBeenCalled()
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`)
  })

  it('should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe(imageUrl)
  })

  it('should ContentTitle have title', () => {
    expect(wrapper.find('ContentTitle').text()).toBe(title.toUpperCase())
  })
})
