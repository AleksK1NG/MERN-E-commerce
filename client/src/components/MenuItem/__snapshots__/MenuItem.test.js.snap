// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render menuItem component 1`] = `ShallowWrapper {}`;
